modules/plugins/wfly-10/src/main/java/org/rhq/modules/plugins/wildfly10/BaseProcessDiscovery.java

```diff
@@ -468,14 +469,59 @@ public DiscoveredResourceDetails discoverResource(Configuration pluginConfig, Re
         managementHostPort.host = hostname;
         managementHostPort.port = port;
-        String key = createKeyForRemoteResource(hostname + ":" + port);
-        String name = buildDefaultResourceName(hostPort, managementHostPort, productType, null);
-        //FIXME this is inconsistent with how the version looks like when autodiscovered
-        String version = productInfo.getProductVersion();
+        hostPort.port = HostConfiguration.DEFAULT_NATIVE_PORT;
+        String key, name, version;
+        // Assume the server is local and follow a similar flow as if it was auto-discovered
+        ASConnection connection = new ASConnection(ASConnectionParams.createFrom(serverPluginConfig));
+        File homeDir = serverPluginConfig.getHomeDir();
+        if (homeDir == null) {
+            homeDir = getHomeDirFromConnection(connection);
+            serverPluginConfig.setHomeDir(homeDir);
+        }
+        if (serverPluginConfig.getBaseDir() == null) {
+            serverPluginConfig.setBaseDir(getBaseDirFromConnection(connection));
+        }
+        if (serverPluginConfig.getConfigDir() == null) {
+            serverPluginConfig.setConfigDir(getConfigDirFromConnection(connection));
+        }
+
+        File hostXmlFile = serverPluginConfig.getHostConfigFile();
+        if (hostXmlFile == null) {
+            hostXmlFile = getHostXmlFileFromConnection(connection);
+            serverPluginConfig.setHostConfigFile(hostXmlFile);
+        }
+        File logDir = serverPluginConfig.getLogDir();
+        if (logDir == null) {
+            logDir = getLogDirFromConnection(connection);
+            serverPluginConfig.setLogDir(logDir);
+        }
+        File logFile = getLogFile(logDir);
+        initLogEventSourcesConfigProp(logFile.getPath(), pluginConfig);
+
+        HostConfiguration hostConfig;
+        try {
+            hostConfig = loadHostConfiguration(hostXmlFile);
+            serverPluginConfig.setApiVersion(hostConfig.getDomainApiVersion());
+            pluginConfig.setSimpleValue("hostXmlFileName", hostXmlFile.getName());
+            key = createKeyForLocalResource(serverPluginConfig);
+            name = buildDefaultResourceName(hostPort, managementHostPort, productType, hostConfig.getHostName());
+            version = getVersion(homeDir, productType);
+            serverPluginConfig.setLocal(true);
+        } catch (Exception exception){
+            LOG.info("Manually imported server marked as [REMOTE]. Unable to load configuration file: [" + hostXmlFile.getPath() + "]", exception);
+            hostPort.isLocal = false;
+            managementHostPort.isLocal = false;
+            serverPluginConfig.setLocal(false);
+            key = createKeyForRemoteResource(hostname + ":" + port);
+            name = buildDefaultResourceName(hostPort, managementHostPort, productType, null);
+            //FIXME this is inconsistent with how the version looks like when autodiscovered
```

Comment:
If this is true, then won't it perhaps prevent stuff like bundle deployments to work correctly? We use the version for a lot of things (such as detecting EAP6/7 difference)

Reply:
Perhaps, It was already like that and i'm not fixing the "remote" EAPs.
I'm only trying to see if we can identify if the server is "local" and trying to get their data fill up. If not, fallback to previous code (this).